libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -95,6 +95,17 @@ type
     # Task for processing non-priority message queue.
     sendNonPriorityTask: Future[void]
 
+  CustomConnectionCallbacks* = object
+    customConnCreationCB*: proc(
+      destAddr: Option[MultiAddress], destPeerId: PeerId, codec: string
+    ): Connection {.gcsafe, raises: [].}
+    peerSelectionCB*: proc(
+      allPeers: HashSet[PubSubPeer],
+      directPeers: HashSet[PubSubPeer],
+      meshPeers: HashSet[PubSubPeer],
+      fanoutPeers: HashSet[PubSubPeer],
+    ): HashSet[PubSubPeer] {.gcsafe, raises: [].}
```

Comment:
To make this `CustomConnectionCallbacks` object easier to read, could you extract the procs to a type? That way The object would look like:
```
CustomConnectionCallbacks* = object
    customConnCreationCB*: someCustomProcTypeHere
    peerSelectionCB*: anotherCustomProcTypeHere
```